### PR TITLE
Fixes health-url cli and backend, improves examples

### PR DESCRIPTION
**Pull request by cbron**
Issue: https://github.com/rancher/rio/issues/750

Examples: 
* `rio run --health-url http://:80 -p 80 nginx`
* `rio run --health-url http://:8080/status --health-initial-delay 10s --health-interval 5s --health-failure-threshold 5 --health-timeout 5s -p 8080 cbron/mybusybox:dev`

### CODE REVIEW

File: cli/cmd/create/health.go [hunk 1]
```diff
@@ -69,19 +68,14 @@ func (c *Create) setHealthCheck(spec *riov1.ServiceSpec) error {
...
 			hc.TCPSocket = &v1.TCPSocketAction{
-				Port: intstr.FromInt(port),
+				Port: intstr.FromString(portStr),
 			}
 		} else {
 			hc.HTTPGet = &v1.HTTPGetAction{
-				Port: intstr.FromInt(port),
-				Host: u.Host,
+				Port: intstr.FromString(portStr),
+				Host: u.Hostname(),
```
Comment (cbron) [inline cli/cmd/create/health.go]: `Host` includes port which doubles it, `Hostname()` removes it

File: cli/cmd/create/create.go [hunk 2]
```diff
@@ -56,7 +56,7 @@ type Create struct {
 	HealthInterval         string            `desc:"Time between running the check (ms|s|m|h)" default:"0s"`
 	HealthSuccessThreshold int               `desc:"Consecutive successes needed to report healthy"`
 	HealthTimeout          string            `desc:"Maximum time to allow one check to run (ms|s|m|h)" default:"0s"`
-	HealthURL              string            `desc:"URL to hit to check health (example: http://localhost:8080/ping)"`
+	HealthURL              string            `desc:"URL to hit to check health (example: http://:8080/ping)"`
```
Comment (StrongMonkey) [inline cli/cmd/create/create.go]: why this change?

File: modules/service/controllers/service/populate/pod/containers.go [hunk 3]
```diff
@@ -87,6 +87,17 @@ func toContainer(containerName string, c *riov1.Container) v1.Container {
...
+		// Norman only takes string for intstr types, but K8s only takes int, so do conversion
+		if c.ReadinessProbe.HTTPGet != nil && c.ReadinessProbe.HTTPGet.Port.StrVal != "" {
+			c.ReadinessProbe.HTTPGet.Port = intstr.FromInt(c.ReadinessProbe.HTTPGet.Port.IntValue())
+		}
+		if c.ReadinessProbe.TCPSocket != nil && c.ReadinessProbe.TCPSocket.Port.StrVal != "" {
+			c.ReadinessProbe.TCPSocket.Port = intstr.FromInt(c.ReadinessProbe.TCPSocket.Port.IntValue())
+		}
+		con.ReadinessProbe = c.ReadinessProbe
+	}
+
```
Comment (StrongMonkey) [inline modules/service/controllers/service/populate/pod/containers.go]: Same for liveness?

File: cli/cmd/create/create.go (hunk 2 above)
Comment (cbron) [reply cli/cmd/create/create.go]: It's misleading to include in example, localhost won't work for most cases:
```The kubelet sends the probe to the pod’s IP address, unless the address is overridden by the optional host field in httpGet. If scheme field is set to HTTPS, the kubelet sends an HTTPS request skipping the certificate verification. In most scenarios, you do not want to set the host field. ```
